### DISCUSSION TITLE
Error: Invalid LatLng object: (undefined, undefined)

### DISCUSSION THREAD

**Discussion by MoussaRiad** [alexandra-c/leaflet-tracking-marker#13] [Q&A] [answered]
**Describe the bug** 
I have added the leaflet-tracking-marker to my project( which is for now a map with simple markers), and followed the same code example hosted in codesandbox [link](https://codesandbox.io/s/react-leaflet-tracking-marker-example-ivlhk?file=/src/App.js) ,
But it doesn't rendering anything ( the hole map), and it showed me in the console this error :
<span style="color:red">
Uncaught Error: Invalid LatLng object: (undefined, undefined)
-    at new LatLng (LatLng.js:32:1)
-    at Object.toLatLng [as latLng] (LatLng.js:123:1)
-    at NewClass.initialize (BaseMarker.js:23:1)
-    at new NewClass (Class.js:22:1)
-    at createMarker (LeafletTrackingMarker.js:42:1)
-    at useMutableLeafletElement (element.js:10:1)
-    at useLayer (layer.js:21:1)
-    at ContainerComponent (component.js:6:1)
-    at renderWithHooks (react-dom.development.js:14985:1)
-    at updateForwardRef (react-dom.development.js:17044:1)
A capture of the Map.js file is attached(the tracker marker in this case is called MovingMarker in the last lines)
Any help will be appreciated
![code](https://user-images.githubusercontent.com/78734256/162236025-38ba2252-5535-4d43-aeae-5d823afb2f6a.png)

**Comment by alexandra-c** [comment, marked as answer]
Hello @MoussaRiad,

``` js
const [currentTrack, setCurrentTrack] = useState({});
```
In my example, the "**currentTrack**" is an object not an array, it represents one coordinate of your marker, latitude and longitude, defined by an object with two properties, "lat" and "lng".  

I then destructure this "**data**" object and send its  "lat" and "lng" values to the tracking marker as an array (see the code snippet bellow)
![image](https://user-images.githubusercontent.com/46926675/164253478-9fa06c7d-8928-43fd-b03a-b0f200aeecd2.png)

But this is just my example, you can handle your data differently in your project as long as you make sure that you are sending  the "position" and "prevPos" to **LeafletTrackingMarker ** in the required format.

``` js
<LeafletTrackingMarker icon={icon} position={[latitude, longitude]} previousPosition={[previousLatitude, previousLongitude]} />
```

**Reply by MoussaRiad** [reply]
@alexandra-c , thank you for the answer,
I have solved the problem just after i posted the issue
